From a75e1456f11725b7a58bc81148a6d6403b2530d2 Mon Sep 17 00:00:00 2001
From: Jan Pokorný <jpokorny@redhat.com>
Date: Dec 01 2017 10:47:51 +0000
Subject: formats/corosync.rng: support nodelist.node.name configuration option


This option within corosync configuration was silently established (and
suggested to overcome certain node identification issues) by pacemaker
without any awareness within the former, and hence was completely
foreign also to a detached attempt to formalize this public config,
interface (known as "corosync.conf" file) by the means of RelaxNG
schema describing its XML equivalent and materialized as an in-tree
corosync.rng file.  This used to be prone to validation issues with the
cib2pcscmd* commands, but since corosync now acknowledges that option
just as any other native one[*], set the schema to follow suit.

[*] https://github.com/corosync/corosync/pull/286

Resolves: <https://bugzilla.redhat.com/1517834>
Signed-off-by: Jan Pokorný <jpokorny@redhat.com>

---

diff --git a/formats/corosync/corosync.rng b/formats/corosync/corosync.rng
index 7b12652..f239da8 100644
--- a/formats/corosync/corosync.rng
+++ b/formats/corosync/corosync.rng
@@ -376,6 +376,19 @@ adjust nodes in the cluster.</a:documentation>
     <element name="node">
       <group>
         <optional>
+          <attribute name="name">
+	    <a:documentation>This optional configuration option
+provides a unified way for the client software (e.g. pacemaker) atop,
+respectively the end users, to guide establishing a nominal
+(self-)identification for each node in case neither respective
+*ringX_addr* specifies a network hostname nor other means are
+available/effective in this process.  Option is not used by Corosync
+itself.</a:documentation>
+            
+            <text/>
+          </attribute>
+        </optional>
+        <optional>
           <attribute name="nodeid">
             <a:documentation>This configuration option is optional when
 using IPv4 and required when using IPv6.  This is a 32bit value
@@ -395,8 +408,10 @@ is reserved and should not be used.</a:documentation>
           </attribute>
         </optional>
         <attribute name="ring0_addr">
-          <a:documentation>This specifies IP address of one of the nodes for particular ring
-as denoted by its number (instead 0, there can be higher numbers).</a:documentation>
+	  <a:documentation>This specifies IP or network hostname
+address of the particular node for particular link as denoted by its
+number (there can be higher numbers instead of 0 that marks the first
+link).</a:documentation>
           
           <text/>
         </attribute>

